core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1175,7 +1175,7 @@ class KafkaApis(val requestChannel: RequestChannel,
     val topicResponses = metadataCache.getTopicMetadata(topics, listenerName,
       errorUnavailableEndpoints, errorUnavailableListeners)
 
-    if (topics.isEmpty || topicResponses.size == topics.size || fetchAllTopics) {
+    if (topics.isEmpty || (topics.sizeCompare(topicResponses) == 0) || fetchAllTopics) {
```

Comment:
FYI reviewers -
Comparison of two iterables. Reduces comparison time complexity from O(this.size + that.size) to O(this.size min that.size)